src/Connection.php

```diff
@@ -1953,20 +1953,34 @@ public function executeUpdate(string $sql, array $params = [], array $types = []
     /**
      * BC layer for a wide-spread use-case of old DBAL APIs
      *
-     * @deprecated This API is deprecated and will be removed after 2022
+     * @deprecated Use {@see executeQuery()} instead
      */
     public function query(string $sql): Result
     {
+        Deprecation::trigger(
+            'doctrine/dbal',
+            'https://github.com/doctrine/dbal/pull/5948',
```

Comment:
I think https://github.com/doctrine/dbal/pull/4163 might be more relevant as to why these methods are being removed.